### PR TITLE
removeTempUser: maintenance script removing TempUser accounts

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/CE-1182 (required by https://wikia-inc.atlassian.net/browse/PLATFORM-1055)
- remove `TempUser%` accounts from a given database (defaults to `wikicities`, can be used on `uncyclo` as well - via `--db` option)
- check the structure of the user name - `TempUser+<user ID>`
- remove accounts not touched in the last year only
- perform deletes in batches of 1000 users to limit the impact on shared DB (load and slave lag)

**70% of all accounts** on `wikicities` are TempUsers and ~10% on uncyclo. Let's finally clean it up :)

Based on @kamilkoterba's https://github.com/Wikia/community-engineering-misc/pull/8

@Wikia/community-engineering / @michalroszka

### CODE REVIEW

File: maintenance/wikia/removeTempUser.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		$db = $this->getOption( 'db', $wgExternalSharedDB );
+		$isDryRun = $this->hasOption( 'dry-run' );
+
+		$this->output( "Removing temp user accounts from '$db'...\n");
+
+		$dbr = wfGetDB( DB_SLAVE, [], $db );
+		$dbw = wfGetDB( DB_MASTER, [], $db );
+
+		// get the list of accounts for remove
+		$users =$this->getTempUserAccounts( $dbr );
```
Comment (michalroszka) [inline maintenance/wikia/removeTempUser.php]: Paste the following space here → ← right after =. :-)

**Comment by michalroszka** [comment]
per our conversation - please, check the secondary clusters

**Comment by macbre** [comment]
@michalroszka, good point - implemented in e554419

**Comment by michalroszka** [comment]
LGTM :+1:

**Comment by Grunny** [comment]
LGTM.

File: includes/specials/SpecialUserlogin.php [hunk 2]
```diff
@@ -1080,14 +1080,7 @@ function mailPasswordInternal( $u, $throttle = true, $emailTitle = 'passwordremi
 
 		$np = $u->randomPassword();
 		$u->setNewpassword( $np, $throttle );
-
-		/* Wikia change begin */
-		//@TODO get rid of TempUser handling when it will be globally disabled
-		$tempUser = null;
-		wfRunHooks( 'MailPasswordTempUser' , array( &$u, &$tempUser ) );
```
Comment (macbre) [inline includes/specials/SpecialUserlogin.php]: This hook was not used anywhere.

**Comment by michalroszka** [comment]
LGTM :+1:

**Pull request merged**